clang/test/Sema/implicit-int-conversion-on-int.c

```diff
@@ -0,0 +1,64 @@
+// RUN: %clang_cc1 %s -verify -Wimplicit-int-conversion
+// RUN: %clang_cc1 %s -verify -Wimplicit-int-conversion -Wno-implicit-int-conversion-on-negation -DNO_DIAG
```

Comment:
A better approach than using `-D` is to specify a prefix to `-verify`: https://clang.llvm.org/docs/InternalsManual.html#verifying-diagnostics

Then you can have one comment for the `-no-diagnostics` RUN line and skip all the preprocessor code for the other RUN line.